__tests__/integration/client/getAllcommitmentsByListOfCompressedZkpPublicKey.test.ts

```diff
@@ -0,0 +1,56 @@
+import * as dotenv from "dotenv";
+import path from "path";
+import { Client } from "../../../libs/client/index";
+import ICommitment from "../../../libs/models/commitment";
+import { User } from "../../../libs/user";
+import { NIGHTFALL_DEFAULT_CONFIG } from "../../../libs/user/constants";
+
+const _rootPath = path.resolve();
+dotenv.config({ path: path.join(_rootPath, ".env") });
+
+const environment = {
+  blockchainNetwork: process.env.SDK_ENV_BLOCKCHAIN_NETWORK,
+  blockchainWsUrl: process.env.SDK_ENV_BLOCKCHAIN_WEBSOCKET_URL,
+  clientApiUrl: process.env.SDK_ENV_CLIENT_API_URL,
+};
+
+describe("Suit of tests for get commitmens from some endpoint", () => {
+  let user: User;
+  let client: Client;
+  let commitmens: ICommitment[];
+
+  beforeAll(async () => {
+    client = new Client(NIGHTFALL_DEFAULT_CONFIG.clientApiUrl);
+    user = new User(environment); // now goerli
+    await user.init({
+      ethereumPrivateKey: process.env.SDK_ETH_PRIVATE_KEY,
+      nightfallMnemonic: process.env.SDK_NIGHTFALL_MNEMONIC,
+    });
+  });
+
+  test("Should get a json from getCommitmentsOnChain endpoint", async () => {
+    commitmens = await client.getCommitmentsByCompressedZkpPublicKey([
+      user.zkpKeys.compressedZkpPublicKey,
+    ]);
+    expect(commitmens).toBeInstanceOf(Object as unknown as ICommitment[]);
+    expect(commitmens[0].preimage.compressedZkpPublicKey).toEqual(
+      user.zkpKeys.compressedZkpPublicKey,
+    );
+  });
+
+  test("Should passs a empty array of strings and receive null", async () => {
+    commitmens = await client.getCommitmentsByCompressedZkpPublicKey([]);
+    expect(commitmens).toBe(null);
+  });
```

Comment:
Shall we allow the developer to get this far? If this is meant to export commitments by a given public key but no key is given... Should we allow them to continue?

On the other hand I'm not sure we need this integration test as it is testing the return from the API (so this test belongs in the API).

Just food for thought. Let's discuss.

Reply:
```
  /**
   *
   * @method getCommitmentsByCompressedZkpPublicKey does the communication with the nightfall client
   * endpoint to get all commitments by compressed pkd.
   * @param listOfCompressedZkpPublicKey a list of compressed zkp publick keys derivated from
   * the user mnemonic.
   * @returns all the commitments existent for this compressed pkds.
   * @author luizoamorim
   */
  async getCommitmentsByCompressedZkpPublicKey(
    listOfCompressedZkpPublicKey: string[],
  ): Promise<Commitment[]> {
    try {
      if (
        listOfCompressedZkpPublicKey &&
        listOfCompressedZkpPublicKey.length > 0
      ) {
        const response = await axios.post(
          `${this.apiUrl}/commitment/compressedZkpPublicKeys`,
          listOfCompressedZkpPublicKey,
        );

        return response.data.commitmentsByListOfCompressedZkpPublicKey;
      }
      throw new Error("You should pass at least one compressedZkpPublicKey");
    } catch (err) {
      logger.child({ listOfCompressedZkpPublicKey }).error(err);
      return null;
    }
  }
}